libexec/swiftenv

```diff
@@ -4,10 +4,14 @@
 set -e
 
 if [ -z "$SWIFTENV_ROOT" ]; then
-  SWIFTENV_ROOT=$HOME/.swiftenv
+  if [[ -n "${XDG_DATA_HOME}" ]] ; then
```

Comment:
In order to make this a backwards compatible change (otherwise an upgrade to swiftenv would result in loosing current swift versions and state). Please alter the behaviour so that, if `$HOME/.swiftenv` exists then it is used, otherwise we use `$XDG_DATA_HOME` (and we could make it the default if there is no prior setup in `~/.swiftenv`).

```bash
if [ -d "$HOME/.swiftenv" ]; then
  SWIFTENV_ROOT="$HOME/.swiftenv"
elif [ -n "$XDG_DATA_HOME" ]; then
  SWIFTENV_ROOT="$XDG_DATA_HOME/.swiftenv"
else
  ...
fi
```